src/Vault.ts

```diff
@@ -0,0 +1,419 @@
+    return this._transactSequence(async function* ({ walletClient, signingAddress, publicClient }) {
+      const { centrifugeRouter } = lpConfig[self.chainId]!
+      const [estimate, investment] = await Promise.all([self.network._estimate(), self.investment(signingAddress)])
+
+      if (investment.pendingInvestCurrency.isZero()) throw new Error('No order to cancel')
+
+      yield* doTransaction('Cancel Invest Order', publicClient, () =>
+        walletClient.writeContract({
+          address: centrifugeRouter,
+          abi: ABI.CentrifugeRouter,
+          functionName: 'cancelDepositRequest',
+          args: [self.address, estimate],
+          value: estimate,
+        })
+      )
+    }, this.chainId)
+  }
+
+  /**
+   * Place an order to redeem funds from the vault. If an order exists, it will increase the amount.
+   * @param shares - The amount of shares to redeem
+   */
+  increaseRedeemOrder(shares: bigint | number) {
+    const self = this
+    return this._transactSequence(async function* ({ walletClient, signingAddress, publicClient }) {
+      const { centrifugeRouter } = lpConfig[self.chainId]!
+      const [estimate, investment] = await Promise.all([self.network._estimate(), self.investment(signingAddress)])
+      const amount = new Token(shares, investment.shareCurrency.decimals)
+
+      if (amount.gt(investment.shareBalance)) throw new Error('Insufficient balance')
+      if (!amount.gt(0n)) throw new Error('Order amount must be greater than 0')
+
+      yield* doTransaction('Redeem', publicClient, () =>
+        walletClient.writeContract({
+          address: centrifugeRouter,
+          abi: ABI.CentrifugeRouter,
+          functionName: 'requestRedeem',
+          args: [self.address, amount.toBigInt(), signingAddress, signingAddress, estimate],
+          value: estimate,
+        })
+      )
+    }, this.chainId)
+  }
+
+  /**
+   * Cancel an open redemption order.
+   */
+  cancelRedeemOrder() {
+    const self = this
+    return this._transactSequence(async function* ({ walletClient, signingAddress, publicClient }) {
+      const { centrifugeRouter } = lpConfig[self.chainId]!
+      const [estimate, investment] = await Promise.all([self.network._estimate(), self.investment(signingAddress)])
+
+      if (investment.pendingRedeemShares.isZero()) throw new Error('No order to cancel')
+
+      yield* doTransaction('Cancel Redeem Order', publicClient, () =>
+        walletClient.writeContract({
+          address: centrifugeRouter,
+          abi: ABI.CentrifugeRouter,
+          functionName: 'cancelRedeemRequest',
+          args: [self.address, estimate],
+          value: estimate,
+        })
+      )
+    }, this.chainId)
+  }
+
+  /**
+   * Claim any outstanding fund shares after an investment has gone through, or funds after an redemption has gone through.
+   * @param receiver - The address that should receive the funds. If not provided, the investor's address is used.
+   */
+  claim(receiver?: string) {
```

Comment:
Would be good to also add an optional `controller` param here, so we can actually use the SDK to build a bot to claim for others